src/NuGet.Clients/NuGet.PackageManagement.UI/Models/PackageDetailControlModel.cs

```diff
@@ -182,13 +200,61 @@ protected override Task CreateVersionsAsync(CancellationToken cancellationToken)
             // Add disabled versions
             AddBlockedVersions(blockedVersions);
 
+            // Update the view and add the filter
+            VersionsView = new CollectionViewSource() { Source = Versions }.View;
+            VersionsView.Filter += VersionsFilter;
+
             SelectVersion();
 
             OnPropertyChanged(nameof(Versions));
+            OnPropertyChanged(nameof(VersionsView));
+
+            // If the selected version is not the first one on the list, the line above will change it to the first one
+            if (_nugetProjects.Any() && 
+                _nugetProjects.FirstOrDefault().ProjectStyle.Equals(ProjectModel.ProjectStyle.PackageReference) &&
+                SelectedVersion != FirstDisplayedVersion)
+            {
+                SelectedVersion = FirstDisplayedVersion;
+                UserInput = SelectedVersion?.ToString();
+            }
 
             return Task.CompletedTask;
         }
 
+        private bool VersionsFilter(object o)
+        {
+            var version = o as DisplayVersion;
+            // If the text is empty or is the insalled version we should show all the versions like if there where no filtering
+            if (string.IsNullOrEmpty(UserInput) || UserInput.Equals(FirstDisplayedVersion?.ToString(), StringComparison.OrdinalIgnoreCase)) return true;
+
+            // Handle of *, that this will show all versions other than the installed version, the suggested version and the null separator
+            if (UserInput.Equals("*", StringComparison.OrdinalIgnoreCase))
+            {
+                if (version == null || version.Version.Equals(Versions[0].Version) || version.Version.Equals(Versions[1].Version)) return false;
+                return true;
```

Comment:
Maybe it's the lack of `{ }` but I'm confused by this block. Why is it important to check the 1st and 2nd version?

Reply:
Hmm maybe this is not the correct implementation, specially when you have preview versions, what Im trying to do is show only the versions that are after the null separator, not sure if using a variable for this is the best option.

Reply:
Ohh, yeah there's no guarantee of having a Latest Stable and a Latest Prerelease. Might be a bug here. 

Could we find the position of 'null' separator, then start looking from there for 0-N versions? (I would assume 0 is possible because I'm pretty sure it can happen during slow perf scenarios)